ios/gamma/split-view/RNSSplitViewAppearanceApplicator.swift

```diff
@@ -0,0 +1,196 @@
+///
+/// @brief - Class responsible for applying all upcoming updates to SplitView.
+///
+/// This class is synchronizing UISplitViewController configuration props which are affecting the SplitView appearance with props passed to RNSSplitViewHostComponentView from the ElementTree.
+///
+class RNSSplitViewAppearanceApplicator {
+  ///
+  /// @brief Function responsible for applying all updates to SplitView in correct order
+  ///
+  /// It requests calling proper callbacks with batched SplitView updates on the AppearanceCoordinator object
+  ///
+  /// @param splitView The view representing JS component which is sending updates.
+  /// @param splitViewController The controller associated with the native JS component which receives updates and manages the native layer.
+  /// @param appearanceCoordinator The coordinator which is checking whether the update needs to be applied and if so, it executes the callback passed by this class.
+  ///
+  public func updateAppearanceIfNeeded(
+    _ splitView: RNSSplitViewHostComponentView?, _ splitViewController: RNSSplitViewHostController?,
+    _ appearanceCoordinator: RNSSplitViewAppearanceCoordinator?
+  ) {
+    precondition(
```

Comment:
[Precondition](https://developer.apple.com/documentation/swift/precondition(_:_:file:line:)) looks a bit scary to me, that it can behave in other ways depending on compiler flags.


Moreover, if we don't accept nullables, maybe it should be forced to pass non-null objects here?

Reply:
https://github.com/software-mansion/react-native-screens/pull/3065/commits/62a6f9f0fa065c68c8bd692ccb8fd713d47b7e98